### DISCUSSION TITLE
STS OpenID multiple claims

### DISCUSSION THREAD

**Discussion by bruno-villa** [minio/minio#12222] [Q&A] [answered]
Hello,

I'm trying to set up a MinIO instance and use the STS to provide access to both human users and other services. I would like to use the authentication and implicit flows for humans and the client credentials flow for services. Individually this works fine. I can have the IdP provide a `roles` claim with the groups that a person is a member of and map those to policies. Alternatively I can use an `application_id` claim obtained via a client credentials grant and map that to a policy. MinIO can be configured to use one of the two by setting `MINIO_IDENTITY_OPENID_CLAIM_NAME` to either `roles` or `application_id`. However, I would like to check both claims for a valid policy. From the mc CLI help it looks like this parameter only accepts a string. Is there a way to specify a list of claims in this string (tried csv, spaces, etc.)? The only other solution I can think of is to configure the IdP to provide the claims under the same name, which is not ideal.

**Comment by harshavardhana** [comment, marked as answer]
> Hello,
> 
> I'm trying to set up a MinIO instance and use the STS to provide access to both human users and other services. I would like to use the authentication and implicit flows for humans and the client credentials flow for services. Individually this works fine. I can have the IdP provide a `roles` claim with the groups that a person is a member of and map those to policies. Alternatively I can use an `application_id` claim obtained via a client credentials grant and map that to a policy. MinIO can be configured to use one of the two by setting `MINIO_IDENTITY_OPENID_CLAIM_NAME` to either `roles` or `application_id`. However, I would like to check both claims for a valid policy. From the mc CLI help it looks like this parameter only accepts a string. Is there a way to specify a list of claims in this string (tried csv, spaces, etc.)? The only other solution I can think of is to configure the IdP to provide the claims under the same name, which is not ideal.

Just use single claims, a single claim can have multiple values.

It's not worth complicating the claim logic here.
